src/pytorch_lightning/lite/wrappers.py

```diff
@@ -115,6 +117,26 @@ def _convert_float_tensor(t: Tensor) -> Tensor:
         output = apply_to_collection(output, function=_convert_float_tensor, dtype=Tensor)
         return output
 
+    @overload
+    def state_dict(self, *, destination: T_destination, prefix: str = ..., keep_vars: bool = ...) -> T_destination:
+        ...
+
+    @overload
+    def state_dict(self, *, prefix: str = ..., keep_vars: bool = ...) -> T_destination:
+        ...
+
+    def state_dict(
+        self, *args: Any, destination: Optional[T_destination] = None, prefix: str = "", keep_vars: bool = False
+    ) -> Optional[T_destination]:
```

Comment:
Why optional?

Reply:
If I don't, mypy infers it can be optional. I'm just trying to override torch's method, which doesn't use mypy. If you have a better suggestion, please modify it.

Reply:
In torch they just don't provide any typing (including return types) as the types should be clear from overloads: https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/module.py#L1374

Reply:
Okay. Not sure how to fix this so resolving